client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/protocol/InternalFlag.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.client.hotrod.impl.protocol;
+
+/**
+ * Flags which are internal to the protocol and shouldn't be exposed to users.
```

Comment:
I like this last proposal but I don't like having optional fields, particularly in the header. We already have to deal with optional parameters due to me adding `vInt` and `vLong` instead of 32bit and 64bit numbers, and that makes it hard to read the entire header size in one go. A very good article on the topic can be found in [Martin Thompson's Mechanical Sympathy blog](http://mechanical-sympathy.blogspot.cz/2014/05/simple-binary-encoding.html). At some point, maybe 3.0, I'd like to redo the protocol following the advice in that blog post which makes a lot of sense.